lib/ahoy_email/processor.rb

```diff
@@ -99,9 +99,10 @@ def track_links
 
         doc = Nokogiri::HTML(body.raw_source)
         doc.css("a[href]").each do |link|
+          uri = Addressable::URI.parse(link["href"])
```

Comment:
> We should probably rescue this (current code also should, but doesn't)

I don't know if that's needed. By looking at [parse](https://github.com/sporkmonger/addressable/blob/master/lib/addressable/uri.rb#L83) source it can fail only when the argument is not a string (or does not respond to `to_str`), it also handle `nil` case.

I think nokogiri always returns strings (or nil) for attributes thus it should be safe.

Random garbage thrown into href attribute will be silently ignored:

```ruby
uri = Addressable::URI.parse '\this\is\not\a\valid\url'
{
      :scheme => nil,
        :user => nil,
    :password => nil,
        :host => nil,
        :port => nil,
        :path => "\\this\\is\\not\\a\\valid\\url",
       :query => nil,
    :fragment => nil
}
uri.absolute? # => false, i.e. trackable?(uri) => false
```